pmd-java/src/test/java/net/sourceforge/pmd/lang/java/oom/DataStructureTest.java

```diff
@@ -30,13 +46,39 @@
  */
 public class DataStructureTest extends ParserTst {
 
+    MetricKey<ClassMetric> classMetricKey = new MetricKey<ClassMetric>() {
+        @Override
+        public String name() {
+            return null;
+        }
+
+
+        @Override
+        public ClassMetric getCalculator() {
+            return new RandomMetric();
+        }
+    };
+    MetricKey<OperationMetric> opMetricKey = new MetricKey<OperationMetric>() {
+        @Override
+        public String name() {
+            return null;
+        }
+
+
+        @Override
+        public OperationMetric getCalculator() {
+            return new RandomMetric();
+        }
+    };
     private PackageStats pack;
 
+
     @Before
     public void setUp() {
         pack = new PackageStats();
     }
 
+
```

Comment:
Yet another metric! You're welcome to make a feature request :)

I used to cramp up my classes too, writing ` { } ` on one line, skipping braces for `if` statements, etc. because I have a small monitor. But I found out it's much easier to skim a class and find info when you can find out if hunks of code are their own methods at a glance: two blank lines => two different methods, one blank line => inside a method. I think I might have a very tubular vision too because anything more than a few cm away from where I look on the monitor is just annoying noise to me, which is why I like to keep things clearly separated

Doing that actually helped me focus on the actual code, which is in the methods (a java class is so much boilerplate after all). For me information density is a relevant metric inside a single method, not so much a class. I totally get that feeling of "mildly perverse pleasure" when I get to write the most elegant method I could, but the more satisfaction I get, the more frustration I feel when Checkstyle bashes me with "assignments in guard clauses are not allowed" or whatever :)

Btw I had never seen the word *grok* before but I'll keep it, thanks!